google/generativeai/types/retriever_types.py

```diff
@@ -33,11 +33,12 @@
 from google.generativeai.utils import flatten_update_paths
 
 
-_DOCUMENT_NAME_REGEX = re.compile(r"^corpora/[a-z0-9-]+/documents/[a-z0-9-]+$")
-_CHUNK_NAME_REGEX = re.compile(r"^corpora/([^/]+?)(/documents/([^/]+?)(/chunks/([^/]+?))?)?$")
-_REMOVE = string.punctuation
-_REMOVE = _REMOVE.replace("-", "")  # Don't remove hyphens
-_PATTERN = r"[{}]".format(_REMOVE)  # Create the pattern
+# _DOCUMENT_NAME_REGEX = re.compile(r"^corpora/[a-z0-9-]+/documents/[a-z0-9-]+$")
+# _CHUNK_NAME_REGEX = re.compile(r"^corpora/([^/]+?)(/documents/([^/]+?)(/chunks/([^/]+?))?)?$")
+# _REMOVE = string.punctuation
+# _REMOVE = _REMOVE.replace("-", "")  # Don't remove hyphens
+# _PATTERN = r"[{}]".format(_REMOVE)  # Create the pattern
+_VALID_NAME = r"^[^-][a-zA-Z\d-][^-]+$"
```

Comment:
Is this different from the other _VALID_NAME in `retriever.py`?

If not, remove one & use one definition.

If they are different, the names should describe this (e.g. `_VALID_CORPUS_NAME` and `_VALID_DOCUMENT_NAME`)